File: submissions/1297-maximum-number-of-balloons/solution.ts
function maxNumberOfBalloons(text: string): number {
   let total:number = Infinity;
   let word:string = "balloon"; 
    let count = new Map();
   for(let i = 0; i < text.length; i++){
       count.set(text[i], (count.get(text[i]) || 0) + 1);
       
   } 
    
 for (let char of word) {
        if (!count.has(char)) {
            return 0;
        }

        total = Math.min(total, Math.floor(count.get(char) / (char === 'l' || char === 'o' ? 2 : 1)));
    }

    return total;
    
};

File: submissions/1297-maximum-number-of-balloons/solution.py
from collections import Counter

class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        countsText = Counter(text)
        countsBalloon = Counter("balloon")
        ans = float('inf') 
        
        for ch, count in countsBalloon.items():
            ans = min(ans, countsText[ch] // count if ch in countsText else 0)
        
        return ans

